doc/source/whatsnew/v1.2.0.rst

```diff
@@ -627,6 +627,7 @@ MultiIndex
 - Bug in :meth:`DataFrame.reset_index` with ``NaT`` values in index raises ``ValueError`` with message ``"cannot convert float NaN to integer"`` (:issue:`36541`)
 - Bug in :meth:`DataFrame.combine_first` when used with :class:`MultiIndex` containing string and ``NaN`` values raises ``TypeError`` (:issue:`36562`)
 - Bug in :meth:`MultiIndex.drop` dropped ``NaN`` values when non existing key was given as input (:issue:`18853`)
+- Bug in :meth:`MultiIndex.intersection` returned duplicates when at least one of the indexes had duplicates (:issue:`36915`)
```

Comment:
xref https://github.com/pandas-dev/pandas/issues/36915#issuecomment-734944112 This is fixing a regression from 1.0.5? Can we target 1.1.5 with this?

Reply:
Moved the whatsnew, but this may be tricky to backport.

Reply:
I've not yet reviewed this PR. are there any changes other than those required to fix the regression here?

Reply:
Merge and setops relied partially on the wrong behavior, had to fix this too.

Reply:
hmm, not investigated fully, but this PR may sit on top of #37171 (which incidentally caused a perf regression https://github.com/pandas-dev/pandas/pull/37171#issuecomment-714445812 so maybe that we want to revert that first before committing the changes here anyhow)

Reply:
> but this may be tricky to backport.

opened #38133 to test

Reply:
not checked all the envs, but a quick check on a couple shows the same tests failing on the backport as here. so that's encouraging.

I think we should target as a backport. and if issues down the road will just have to move the release note back.

Reply:
Yep, thats a good sign. Lets try this then, when everything is green here